### PR TITLE
Combined PR for updating data in the API, providing JSON endpoints for that updated data, and functionality for caching the API in Varnish

### LINKED ISSUE #78: New endpoint for returning JSON object containing all places, HUCs, and protected areas
An endpoint that will return a JSON object of all places (Alaska only at first), HUCS, and protected areas such that apps (IEM only thus far) can get an up-to-date account of those via a process run and cached from the API.

### LINKED ISSUE #79: New endpoint for populating the cache with all expected values
An endpoint (not displayed on the API templates as a path) that will re-populate the cache on visiting it. This means firing off a bunch of GET requests to the site itself to ensure all of the cached elements are always fast as can be.

### PR DESCRIPTION
**Pull request by BobTorgerson**
This PR provides new functionality to the API with several new endpoints.

1.  **/update**
This endpoint will go out to the https://github.com/ua-snap/geospatial-vector-veracity repository to pull down the latest version of the Alaska location's CSV, and the shapefile data for the HUCs and protected areas. Once downloaded, it runs Python code to convert those files into JSON files that can be requested via the next endpoint.

2. **/places**
This endpoint requests a JSON representation of the data collected from the endpoint above. These endpoints should mirror the ones currently found in the IEM webapp: https://github.com/ua-snap/iem-webapp/tree/main/assets

The endpoints available are: /places/communities, /places/hucs, /places/protected_areas, and /places/all

3. **/cache**
**NOTE:** This endpoint should only be run on the Varnish cache!  
This will query all desired endpoints to pre-populate our cache. It takes the routes from the Flask app and modifies the returned routes to use GET requests to query all communities, HUCs, and protected areas. For any GET request that does not return a 200 HTTP status code, it logs the status code along with the URL that caused it.

Closes #78 
Closes #79

### CODE REVIEW

**Review by brucecrevensten — CHANGES_REQUESTED** [review]
This is great!  I have a couple specific notes and then one high-level thought, would using git submodules instead of doing HTTP fetches work here?  That might be a good way to keep things in sync with less code, but I don't know.  Also can you make a combined `places/all` route?

File: routes/vectordata.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    # Generates path to JSON
+    jsonpath = f"data/jsons/{json_types[type]}"
+
+    # If the JSON doesn't exist, it needs be generated.
+    if not os.path.exists(jsonpath):
+        update_data()
+
+    # Open JSON file and return to requestor
+    df = pd.read_json(jsonpath)
+    return df.to_json(orient="records")
```
[the remark this answers is not in the document]
Comment (BobTorgerson) [reply routes/vectordata.py]: This has been updated to use the Response() for Flask with MIMETYPE application/json.

File: routes/vectordata.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+            # Adds type to JSON of 'huc'
+            hucs_json[key]["type"] = "huc"
+
+            # Append the JSON object to end of list.
+            output.append(hucs_json[key])
+
+        # Dump output list into local JSON file
+        with open(f"{jsonpath}ak_hucs.json", "w") as outfile:
+            json.dump(output, outfile)
```
[the remark this answers is not in the document]
Comment (BobTorgerson) [reply routes/vectordata.py:163]: These files have been removed from the Git repository and have been added to the .gitignore file.

File: routes/vectordata.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            None.
+
+        Returns:
+            Web page response indicating if a successful update of the data
+            took place.
+
+        Notes:
+            example: http://localhost:5000/update
+    """
+    if update_data():
```
[the remark this answers is not in the document]
Comment (BobTorgerson) [reply routes/vectordata.py]: Removed conditional response from update_data() and now it doesn't return anything.

File: templates/vectordata/failed_update.html [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{% extends 'base.html' %}
```
[the remark this answers is not in the document]
Comment (BobTorgerson) [reply templates/vectordata/failed_update.html]: Removed both templates as there are no longer conditional responses. It either errors out if fail, or returns a 200 response with a simple JSON response.

File: routes/vectordata.py [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+# local imports
+from . import routes
+from luts import json_types
+
+data_api = Blueprint("data_api", __name__)
+
+
+@routes.route("/places/<type>")
+def get_json_for_type(type):
```
[the remark this answers is not in the document]
Comment (BobTorgerson) [reply routes/vectordata.py]: The /places/all route has been created and will return a combined JSON object of all the communities, HUCs, and protected areas. The JSON fields are differentiated by a key of "type" which is one of three values: ['community', 'huc', or 'protected_area']

File: routes/vectordata.py [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+        Returns:
+            JSON-formatted output of all communities, HUCs,
+            or protected areas.
+
+        Notes:
+            example: http://localhost:5000/places/communities
+    """
+    # Generates path to JSON
+    jsonpath = f"data/jsons/{json_types[type]}"
```
[the remark this answers is not in the document]
Comment (BobTorgerson) [reply routes/vectordata.py]: This has been done by adding the full paths to the luts.py file and querying based on the type variable.

**Pull request merged**